### ISSUE TITLE
Acceptance test undefined index TwoFactorTOTPContext.php line 103

### ISSUE THREAD

**Issue by phil-davis** [owncloud/twofactor_totp#118]
Sometimes (rarely, I think) the following can hasppen in the acceptance tests:
https://drone.owncloud.com/owncloud/twofactor_totp/215/144
```
  Scenario: The secret code from qr code should match with the one displayed in the page 2 # /var/www/owncloud/apps/twofactor_totp/tests/acceptance/features/webUITwoFactorTOTP/keyVerification2.feature:19
    Given user "newly-created-user" has logged in using the webUI                          # WebUILoginContext::theUserLogsInUsingTheWebUI()
    And the user has browsed to the personal security settings page                        # WebUIPersonalSecuritySettingsContext::theUserBrowsesToThePersonalSecuritySettingsPage()
    When the user activates TOTP Second-factor auth but does not verify                    # TwoFactorTOTPContext::theUserHasActivatedTOTPSecondFactorAuthButNotVerified()
    Then the secret code from QR code should match with the one displayed on the webUI     # TwoFactorTOTPContext::theSecretCodeFromQRCodeShouldMatchWithTheOneDisplayedOnTheWebUI()
      Notice: Undefined index: secret in /var/www/owncloud/apps/twofactor_totp/tests/acceptance/features/bootstrap/TwoFactorTOTPContext.php line 103
SCENARIO RESULT: (fail)
```

I noticed this happen once when running tests repeatedly in PR #115

**Comment by skshetry** [comment]
It's weird given that we wait for the QR image to appear, and parse it ourself. My guess would be, that the QR image element has loaded, but the `src` attribute hasn't yet.

Have you seen this kind of errors lately? If not, we could revisit this later and close for now.

**Comment by phil-davis** [comment]
I have not noticed failures in the nightly runs. So I guess this is not happening these days.
If you can't see an obvious way why this can happen, and how we could adjust the acceptance test code, then close this issue.

**Comment by skshetry** [comment]
I checked the TwoFactorTOTP page. When the **Activate TOTP** is checked, the image and all other settings get appended at once. We are waiting for the image to appear, so, unless the `QRReader` parser is to blame, that is unlikely to happen. And, it happened only once, so, I'd not waste more time into this. If we see this often, let's reopen and investigate.

For the record, if this happens again, dump the contents of `$qrCode->text()` and give a nice error message on what was received (hex value) and what was parsed.

https://github.com/owncloud/twofactor_totp/blob/a76afb8e1f915bbed4292303c6be40e5a30a31de/tests/acceptance/features/bootstrap/TwoFactorTOTPContext.php#L148-L150

Also, wait till the `src` attribute is not empty before the following line (this never happens as I wrote above, but might make tests more rigid).
https://github.com/owncloud/twofactor_totp/blob/a76afb8e1f915bbed4292303c6be40e5a30a31de/tests/acceptance/features/lib/PersonalSecuritySettingsPageWithTOTPEnabled.php#L63


Closing for now.
